xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp

```diff
@@ -260,18 +263,26 @@ void CLinuxRendererGLES::LoadPlane(CYuvPlane& plane, int type,
   glBindTexture(m_textureTarget, plane.id);
 
   // OpenGL ES does not support strided texture input.
+  GLint pixelStore = -1;
   if (stride != static_cast<int>(width * bps))
   {
-    unsigned char* src = static_cast<unsigned char*>(data);
-    for (unsigned int y = 0; y < height; ++y, src += stride)
-    {
-      glTexSubImage2D(m_textureTarget, 0, 0, y, width, 1, type, GL_UNSIGNED_BYTE, src);
-    }
-  }
-  else
-  {
-    glTexSubImage2D(m_textureTarget, 0, 0, 0, width, height, type, GL_UNSIGNED_BYTE, pixelData);
+#if HAS_GLES >= 3
```

Comment:
I may be wrong but you can compile with GLES 3 headers and use a GLES 2 device (Android) - so you still need to check GLES version at run time.

Reply:
Thx!

Reply:
Or perhaps check for `GL_EXT_unpack_subimage` extension present? It looks like some 2.0 devices implement GL_UNPACK_ROW_LENGTH and may support it: https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_unpack_subimage.txt

Reply:
I have not found any device which supports it, so I guess for the most majority its ok.
What do you think?

Reply:
Yeah, may be safer to assume that GL ES 3.0 is required.

Reply:
I'll add if someone comes around with that...